conanfile.py

```diff
@@ -1,132 +1,70 @@
+        tools.get("https://github.com/xiph/vorbis/archive/v%s.tar.gz" % self.version)
+        os.rename("vorbis-%s" % self.version, self.source_subfolder)
 
-        sln_folder = os.path.join(self.source_subfolder, "win32", "VS2010")
-        sln_filename = update_projects_in_solution(sln_folder, self.options.shared)
-
-        with tools.chdir(sln_folder):
-            msbuild = MSBuild(self)
-            msbuild.build(sln_filename, platforms={"x86": "Win32"})
-
-    def build_with_autotools(self):
-        env = AutoToolsBuildEnvironment(self)
-
-        with tools.chdir(self.source_subfolder):
-            with tools.environment_append(env.vars):
-
-                if self.settings.compiler == "clang" and self.settings.arch == "x86":
-                    # http://lists.llvm.org/pipermail/llvm-bugs/2015-November/043264.html
-                    tools.replace_in_file("configure", " -mno-ieee-fp ", " ")
-
-                configure_args = ["--prefix=%s" % self.package_folder]
-
-                if self.options.shared:
-                    configure_args.extend(["--disable-static", "--enable-shared"])
-                else:
-                    configure_args.extend(["--disable-shared", "--enable-static"])
-                env.configure(args=configure_args)
-                env.make()
-                env.make(args=["install"])
 
     def build(self):
-        if self.settings.compiler == "Visual Studio":
-            self.build_with_visual_studio()
-        else:
-            self.build_with_autotools()
+        cmake = CMake(self)
+        if self.settings.os != "Windows":
+            cmake.definitions["CMAKE_POSITION_INDEPENDENT_CODE"] = self.options.fPIC
+        cmake.configure()
+        cmake.build()
 
     def package(self):
-        self.copy("FindOGG.cmake")
-        self.copy("COPYING", src=self.source_subfolder, dst="licenses", keep_path=False)
-        self.copy("LICENSE.md", dst="licenses", keep_path=False)
+        self.copy("FindVORBIS.cmake", ".", ".")
+        self.copy("include/vorbis/*", ".", "%s" % (self.source_subfolder), keep_path=True)
+        self.copy("%s/copying*" % self.source_subfolder, dst="licenses",  ignore_case=True, keep_path=False)
 
         if self.settings.compiler == "Visual Studio":
-            # Visual Studio build is not installing installing any artifact.
-            # Packaging needs to be done.
-            src_include_dir = os.path.join(self.source_subfolder, "include")
-            self.copy("*.h", dst="include", src=src_include_dir, keep_path=True)
+            if self.options.shared:
+                self.copy(pattern="*.dll", dst="bin", keep_path=False)
             self.copy(pattern="*.pdb", dst="bin", keep_path=False)
             self.copy(pattern="*.lib", dst="lib", keep_path=False)
-
+        else:
             if self.options.shared:
-                self.copy(pattern="*.dll", dst="bin", keep_path=False)
+                if self.settings.os == "Macos":
```

Comment:
why not cmake.install() ?

Reply:
On packages I prepared a long time ago, I leave this custom copying until I really need to change it for some reason, so as to avoid bugs popping up. The custom copying sometimes contains tested wisdom amassed during the years. But I don't know if you agree with this strategy. Note that this package was first created upon the first release of conan.

Reply:
I agree!